File: src/main/java/net/hectus/neobb/turn/person_game/block/PTWhiteWool.java
package net.hectus.neobb.turn.person_game.block;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.Turn;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import net.hectus.neobb.turn.person_game.categorization.AttackCategory;
import net.hectus.neobb.turn.person_game.warp.PTSnowWarp;
import org.bukkit.block.Block;

import java.util.List;

public class PTWhiteWool extends BlockTurn implements AttackCategory {
    public PTWhiteWool(NeoPlayer player) { super(player); }
    public PTWhiteWool(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public double damage() {
        return player.game.warp() instanceof PTSnowWarp ? 2 : 4;
    }

    @Override
    public List<Class<? extends Turn<?>>> counteredBy() {
        return List.of(PTGoldBlock.class, PTDripstone.class);
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/person_game/block/PTWhiteWool.kt
package net.hectus.neobb.modes.turn.person_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.Turn
import net.hectus.neobb.modes.turn.default_game.block.BlockTurn
import net.hectus.neobb.modes.turn.person_game.categorization.AttackCategory
import net.hectus.neobb.modes.turn.person_game.warp.PTSnowWarp
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block
import kotlin.reflect.KClass

class PTWhiteWool(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), AttackCategory {
    override val damage: Double
        get() = if (player!!.game.warp is PTSnowWarp) 2.0 else 4.0

    override fun counteredBy(): List<KClass<out Turn<*>>> {
        return listOf(PTGoldBlock::class, PTDripstone::class)
    }
}
